test/unit/models/TagFollow.test.js

```diff
@@ -0,0 +1,155 @@
+import root from 'root-path'
+const setup = require(root('test/setup'))
+const factories = require(root('test/setup/factories'))
+
+describe('TagFollow', () => {
+  var tag, community, user
+
+  beforeEach(function (done) {
+    return setup.clearDb().then(() => {
+      tag = factories.tag()
+      community = factories.community()
+      user = factories.user()
+    })
+    .then(() => Promise.join(
+      tag.save(),
+      community.save(),
+      user.save())
+    .then(() => done()))
+  })
+
+  describe('#toggle', () => {
+    it("creates a TagFollow when there isn't one", () => {
+      const attrs = {
+        tag_id: tag.id,
+        user_id: user.id,
+        community_id: community.id
+      }
+      return TagFollow.toggle(tag.id, user.id, community.id)
+      .then(() => TagFollow.where(attrs).fetch())
+      .then(tagFollow => {
+        expect(tagFollow).to.exist
+      })
+    })
+
+    it('deletes a TagFollow when there is one', () => {
+      const attrs = {
+        tag_id: tag.id,
+        user_id: user.id,
+        community_id: community.id
+      }
+      return new TagFollow(attrs).save()
+      .then(() => TagFollow.toggle(tag.id, user.id, community.id))
+      .then(() => TagFollow.where(attrs).fetch())
+      .then(tagFollow => {
+        expect(tagFollow).not.to.exist
+      })
+    })
+  })
+
+  describe('#subscribe', () => {
+    it("creates a TagFollow when there isn't one only if isSubscribing", () => {
+      const attrs = {
```

Comment:
this assignment should be pulled out into a `beforeEach` because you use it several times